chain/epoch-manager/src/tests/mod.rs

```diff
@@ -2808,3 +2809,333 @@ fn test_verify_chunk_state_witness() {
         bad_signer.sign_chunk_state_witness(&chunk_state_witness.inner).0;
     assert!(!epoch_manager.verify_chunk_state_witness_signature(&chunk_state_witness).unwrap());
 }
+
+/// Simulate the blockchain over a few epochs and verify that possible_epochs_of_height_around_tip()
+/// gives the correct results at each step.
+/// Some of the blocks are missing to make the test more interesting.
+/// The blocks present in each epoch are:
+/// Epoch(111): genesis
+/// Epoch(111): 1, 2, 3, 4, 5
+///     epoch1: 6, 7, 8, 9, 10
+///     epoch2: 12, 14, 16, 18, 20, 22, 24, 25, 26
+///     epoch3: 27, 28, 29, 30, 31
+///     epoch4: 32+
+#[test]
+fn test_possible_epochs_of_height_around_tip() {
+    use std::str::FromStr;
+
+    let amount_staked = 1_000_000;
+    let account_id = AccountId::from_str("test1").unwrap();
+    let validators = vec![(account_id, amount_staked)];
+    let h = hash_range(50);
+
+    let genesis_epoch = EpochId(CryptoHash::default());
+
+    let epoch_length = 5;
+    let mut epoch_manager = setup_default_epoch_manager(validators, epoch_length, 1, 2, 2, 90, 60);
+
+    // Add the genesis block with height 1000
+    let genesis_height = 1000;
+    record_block(&mut epoch_manager, CryptoHash::default(), h[0], genesis_height, vec![]);
+
+    let genesis_tip = Tip {
+        height: genesis_height,
+        last_block_hash: h[0],
+        prev_block_hash: CryptoHash::default(),
+        epoch_id: genesis_epoch.clone(),
+        next_epoch_id: genesis_epoch.clone(),
+    };
+
+    assert_eq!(
+        epoch_manager.possible_epochs_of_height_around_tip(&genesis_tip, 0).unwrap(),
+        vec![]
+    );
+    assert_eq!(
+        epoch_manager
+            .possible_epochs_of_height_around_tip(&genesis_tip, genesis_height - 1)
+            .unwrap(),
+        vec![]
+    );
+    assert_eq!(
+        epoch_manager.possible_epochs_of_height_around_tip(&genesis_tip, genesis_height).unwrap(),
+        vec![genesis_epoch.clone()]
+    );
+    assert_eq!(
+        epoch_manager
+            .possible_epochs_of_height_around_tip(&genesis_tip, genesis_height + 1)
+            .unwrap(),
+        vec![genesis_epoch.clone()]
+    );
+    assert_eq!(
+        epoch_manager.possible_epochs_of_height_around_tip(&genesis_tip, 10000000).unwrap(),
+        vec![genesis_epoch.clone()]
+    );
+
+    let epoch1 = EpochId(h[0]);
+    dbg!(&epoch1);
```

Comment:
it prints `<file location> <variable name> = <variable value>`, kinda like `#define D(x) std::cout << #x << " = " << x << std::endl;`. A nice convenience macro to quickly print a variable. I left them in because they make debugging the test esasier. Here's how it looks like:
```
running 1 test
[chain/epoch-manager/src/tests/mod.rs:2875:5] &epoch1 = EpochId(
    CoRutESHXR94goeNsP5Za7RKyKrK8AYLQa4PwvT2934w,
)
[chain/epoch-manager/src/tests/mod.rs:2880:9] height = 1001
[chain/epoch-manager/src/tests/mod.rs:2880:9] height = 1002
[chain/epoch-manager/src/tests/mod.rs:2880:9] height = 1003
[chain/epoch-manager/src/tests/mod.rs:2880:9] height = 1004
[chain/epoch-manager/src/tests/mod.rs:2880:9] height = 1005
```

Reply:
Why not use tracing? It's used throughout the codebase and everyone is familiar with it. This is a mini nit given it's only a test so - optional :)

Reply:
Using tracing requires initialising a subscriber which listens to messages and prints them out to the console.
I didn't want to bother with that, so I just used the `dbg!()` which is quick and easy.

But okay, I figured out how to do it (had to add a dependency on `near-o11y`), and the test now uses a subscriber